### DISCUSSION TITLE
Importing GitLab repo to GitHub

### DISCUSSION THREAD

**Discussion by aref1n20** [community/community#53619] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello,
I am trying to import our GitLab repository from GitLab to GitHub using the import option of GitHub. There are a total of 305 commits in the repo and 135 are mine. But when I import the repo, it shows that all the commits are mine. I made sure that everyone has added their GitLab emails as their secondary email in GitHub. What can I do to make sure every contribution is added to the right person

**Comment by mdnurulmomen** [comment, marked as answer]
There are a few options.

Firstly, You may [import ](https://documentation.its.umich.edu/node/4001) repositories from GitLab directly. 

Secondly, If you don't want to hamper your GitLab repositories, you may create a [copy ](https://open.win.ox.ac.uk/pages/open-science/community/Open-WIN-Community/docs/gitlab/6-1-mirroring-to-github/) at your GitHub.

**Reply by mdnurulmomen** [reply]
Please, Let us know if the given answer helped you out or there are still other confusions.

**Reply by aref1n20** [reply]
The first method didn't work before but worked now. Still there is an issue. The commits are showing in my teammates' names but their GitHub account is not linked to the commits. How can i fix it? If you go to my Homeiser repo's commits, you can see what I mean.

**Reply by mdnurulmomen** [reply]
The repo is not showing any contributors ! 
Only, branches and author name are visible. If you don't want to show their name, You may re-clone / mirror protected branch only.

**Reply by aref1n20** [reply]
No I want to show contributors. I made sure everyone has their university email added in GitHub too because their GitLab is opened using the uni email. Still when I import, their GitHub is not getting added to the contributions

**Comment by tluocs** [comment]
I am trying to import a GitLab repo to GitHub but, after a long time of importing (about several minutes), encountered this error:
<img width="653" alt="Screenshot 2024-01-11 at 11 43 28 AM" src="https://github.com/community/community/assets/25195389/b040fe19-b32f-480f-876f-9a6f0663aa75">
I have tried "Restart" twice and the error remained the same (after several minutes' importing). Any thoughts?

I have also tried the "contact support" link, but obviously it doesn't work - it simply points back to this discussion forum.

**Reply by aref1n20** [reply]
maybe check the second answer (295 upvotes) and its comments from this link

https://stackoverflow.com/questions/22265837/transfer-git-repositories-from-gitlab-to-github-can-we-how-to-and-pitfalls-i

**Reply by tluocs** [reply]
### Thank you @Arfii20 - That's exactly what I did, and now I am doing it again:
<img width="736" alt="import" src="https://github.com/community/community/assets/25195389/7930f85c-487c-4cb4-8589-7cbd574ca178">

### But it still failed (after several minutes) with the same error:
<img width="568" alt="Screenshot 2024-01-15 at 2 21 38 PM" src="https://github.com/community/community/assets/25195389/64747bc4-1fae-478d-8ab5-1de2fc00cae7">

**Reply by aref1n20** [reply]
Maybe check if you have appropriate permissions in GitLab or try contacting support.

**Comment by morozovic** [comment]
Hi,

Following the steps from https://stackoverflow.com/questions/22265837/transfer-git-repositories-from-gitlab-to-github-can-we-how-to-and-pitfalls-i, I was able to migrate several projects from GitLab to GitHub with no problem - even with LFS files.

(Well, I assume that from what I've seen, the following format is correct for LFS on GitHub:
```
https://git-lfs.github.com/spec/v1
oid sha256:XYZ
size 123456789
```
is correct.)

However, migrating one project (via the GitHub GUI, from stackoverflow link) I receive an email that it has been successful, but when I want to check the repository, the only thing I see is:

![image](https://github.com/user-attachments/assets/618a33df-d855-47a7-8f7c-373ac310585e)

The project has only the main branch and it was created and only managed by me (same as the other projects which were migrated without issues).

The only difference is that this project is 6,9 GiB in size, where the LFS take around 5 GiB of total space. Could this be the reason why I see nothing in the GitHub repository - having the original repository too large? It is the only thing I could think of, otherwise, I really don't understand why this one migration won't proceed.
